src/noshow/dashboard/helper.py

```diff
@@ -184,3 +189,35 @@ def navigate_patients(
         if st.session_state["name_idx"] > 0:
             st.session_state["name_idx"] -= 1
             st.session_state["pred_idx"] = 0
+
+
+def search_number(
+    Session: sessionmaker, phone_number: str, patient_ids: list[str]
+) -> None:
+    """Search for a patient by phone number
+
+    Parameters
+    ----------
+    Session : sessionmaker
+        SQLAlchemy sessionmaker object
+    phone_number : str
+        Phone number to search for
+    patient_ids : list[str]
+        List of patient ids
+    """
+    with Session() as session:
+        patient_id = session.execute(
+            select(ApiSensitiveInfo.patient_id)
+            .where(
+                (ApiSensitiveInfo.mobile_phone == phone_number)
+                | (ApiSensitiveInfo.home_phone == phone_number)
+                | (ApiSensitiveInfo.other_phone == phone_number)
+            )
+            .distinct()
+        ).scalar()
+
+        if patient_id and patient_id in patient_ids:
```

Comment:
Yes I wanted a simple search function, and while your solution is better, it would also require a bigger change to the code. I'll change the error message